### ISSUE TITLE
nxos_ospf_interfaces: no ip ospf passive-interface is never issued.

### ISSUE THREAD

**Issue by allenrobel** [ansible-collections/cisco.nxos#264]
##### SUMMARY

On Nexus9000, 'ip ospf passive-interface' CLI includes three options (True, False, default), but nxos_ospf_interfaces accepts only two options (True, False).  That is, the following are all valid on Nexus9000:

```
interface Ethernet1/35
  ip ospf passive-interface
  no ip ospf passive-interface
  default ip ospf passive-interface
```

The 'no' form of the command is intended to be used to override the following global 'router ospf' configuration on specific interfaces:

```
router ospf 100
  passive-interface default
```

There are two repercussions to this.

1.  Once 'ip ospf passive-interface' is issued on an interface, it cannot be removed (this requires nxos_ospf_interfaces to support a 'default' value for passive_interface, which it currently does not).

2. The 'no' form of this CLI is never issued, if 'ip ospf passive-interface' is not already configured on the interface.

Issue 1, will require that 'default' is added as a valid value for passive_interface.

Issue 2, will require a change in behavior, such that when the module issues the following:

```
show running-config | section '^interface'
```

It does not interpret a lack of 'ip ospf passive-interface' to mean that its job is done in the case where the user has set passive_interface to false.  Instead, it should issue 'no ip ospf passive-interface'

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

nxos_ospf_interfaces

##### ANSIBLE VERSION

```
ansible 2.10.3
  config file = /Users/arobel/.ansible.cfg
  configured module search path = ['/Users/arobel/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/arobel/ansible-latest/lib/python3.9/site-packages/ansible
  executable location = /Users/arobel/ansible-latest/bin/ansible
  python version = 3.9.0 (default, Oct 27 2020, 14:15:17) [Clang 12.0.0 (clang-1200.0.32.21)]
```

##### CONFIGURATION

```
CACHE_PLUGIN_TIMEOUT(/Users/arobel/.ansible.cfg) = 86400
DEFAULT_LOG_PATH(/Users/arobel/.ansible.cfg) = /tmp/ansible.log
HOST_KEY_CHECKING(/Users/arobel/.ansible.cfg) = False
```

##### OS / ENVIRONMENT

```
NXOS: 9.3(6)
```
##### STEPS TO REPRODUCE

1. Start with the following configuration on the remote device.

```
feature ospf
router ospf 100
  passive-interface default
interface Ethernet1/35
  default ip ospf passive-interface
```

2. Run the following playbook task:

```yaml
    tasks:
    -   cisco.nxos.nxos_ospf_interfaces:
            config:
            -   address_family:
                -   afi: ipv4
                    passive_interface: false
                name: Ethernet1/35
            state: merged
```

Below is a workaround in which the user first enables passive-interface and then disables it.

```yaml
    tasks:
    -   cisco.nxos.nxos_ospf_interfaces:
            config:
            -   address_family:
                -   afi: ipv4
                    passive_interface: true
                -   afi: ipv4
                    passive_interface: false
                name: Ethernet1/35
            state: merged
```

##### EXPECTED RESULTS

1. 'default' should be a valid value for passive_interface
2. When an interface does not contain any form of 'ip ospf passive-interface', the 'no' form of the CLI SHOULD be issued on the interface if that is what the user has requested.

##### ACTUAL RESULTS

1. Running a Playbook with passive_interface set to 'default' results in an error:

```
fatal: [tor-301]: FAILED! => {"ansible_facts": {"discovered_interpreter_python": "/usr/bin/python"}, "changed": false, "msg": "argument passive_interface is of type <type 'str'> found in 'config -> address_family'. and we were unable to convert to bool: The value 'default' is not a valid boolean.  Valid booleans include: 0, 'on', 'f', 'false', 1, 'no', 'n', '1', '0', 't', 'y', 'off', 'yes', 'true'"}
```

2. 'no ip ospf passive-interface' is never issued for an interface, if that interface does not already have some form of 'ip ospf passive-interface' already configured under it.

**Comment by platu** [comment]
Hello,
You are right. The two issues should be merged. Thanks.

**Comment by NilashishC** [comment]
@allenrobel @platu The patch in #290 should fix this issue. Please note that the `passive_interface` key is of type bool and hence cannot take `default` as an option. However, I added a `default_passive_interface` key to get past that. When set to true, if there is any form of "passive-interface" command in running-config, the module will execute "default ip ospf passive-interface" or "default ospfv3 passive-interface".

**Comment by allenrobel** [comment]
@NilashishC  Have merged #290 into my setup and tested.  This looks good from my perspective.  Thank you very much for this!
